packages/router-core/src/router.ts

```diff
@@ -1443,7 +1443,7 @@ export class RouterCore<
       // If it's already a success, update headers and head content
       // These may get updated again if the match is refreshed
       // due to being stale
-      if (match.status === 'success') {
+      if (match.status === 'success' || match.status === 'notFound') {
```

Comment:
while this will invoke the meta fns, the types of those functions are now incorrect I think, as loaderdata is not necessarily defined

Reply:
Thanks @schiller-manuel, may you please help with the correct fix for this? We're keen to update to the latest version but this is a blocker for us 🙏🏿

Reply:
I am not sure. just making the loader data optional would be the easiest ...
@SeanCassiere  why don't we pass in the full `AssetFnContextOptions` into `headers`?

Reply:
If it always updated the meta tags (success or error) that'd be great! So it should update for `success`, `error` and `notFound` but not `redirected` or `pending`?

Reply:
probably? unless there is a reason not to update them?

Reply:
Nope, will make those changes and thanks for the speedy response!